### ISSUE TITLE
db_instance_id is returning the DB Resource ID

### ISSUE THREAD

**Issue by jwoytek** [terraform-aws-modules/terraform-aws-rds#498]
## Description

I am using this module as part of a module in our project's infrastructure. I recently began adding some automated monitoring and was trying to set up some alarms for some RDS metrics. When I attempted to have the module output the `db_instance_id`, I instead get the `db_resource_id` returned. 

The console indicates that an appropriate identifier is set, and requesting the arn for the rds instance shows the identifier correctly. 

- [X] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: 5.9.0

- Terraform version: 1.4.6

- Provider version(s): 5.1.0

## Reproduction Code [Required]
```
output "db_instance_id" {
  value = module.db.db_instance_id
}
```

1. Deploy RDS using module
2. Request `db_instance_id` of the deployed database

## Expected behavior
`db_instance_id` should return the instance identifier and not the resource identifier.

## Actual behavior
`db_instance_id` returns the resource identifier.

### Terminal Output Screenshot(s)
Example (partially redacted):
```
app_db_id = "db-63B5TPHO4P6G2NRAVXEEXXXXXX"
oauth_db_id = "db-IC6E5RWM5UD6OOYMFTKIXXXXXX"
```

## Additional context
Note that another infrastructure deployment operating with provider version 4.58.0 and module version 5.6.0 returns the instance identifier correctly.

**Comment by jwoytek** [comment]
I may have found a thread to pull on here: 

https://github.com/hashicorp/terraform-provider-aws/issues/507

This ticket in the AWS provider is to allow users to change the DB identifier without destroying the existing resource. It seems there are some changes in a commit referenced there that have the module use the resource identifier instead of the instance identifier in some cases. I haven't parsed this fully yet, but this might be part of the issue.

**Comment by jwoytek** [comment]
OK, I believe I have found the issue.

https://github.com/dee-kryvenko/terraform-provider-aws/commit/7eed98333cb3ca9cf45171d822c1a159d1ac7ab1#diff-ca7f85b82bb2b9a4d5dcceea9a2f6a17fa978071f44c026f954e005598c61eecR1349

The `id` field of aws_db_instance is now returning the resource ID, due to the change in the previously referenced ticket. 

This line:
https://github.com/dee-kryvenko/terraform-provider-aws/commit/7eed98333cb3ca9cf45171d822c1a159d1ac7ab1#diff-ca7f85b82bb2b9a4d5dcceea9a2f6a17fa978071f44c026f954e005598c61eecR1420

... appears to create a new and currently undocumented attribute named `identifier` that will return the actual instance identifier. 

I do not have the ability to test this at the moment, but it appears that in addition to the issue in this module, there is probably also a documentation update that needs to be made on the provider end.

**Comment by jwoytek** [comment]
Ugh I missed that this is the same as  #495 (and handled by PR #496) because my search was apparently not awesome enough.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 6.0.0 :tada:
